### DISCUSSION TITLE
Select component with option to create related record

### DISCUSSION THREAD

**Discussion by raftalks** [filamentphp/filament#1758] [Ideas]
I came across following scenario, and thought it would be very useful to have an option in the BelongsToSelect form field which can allow create a record by showing a model of the related resource, when there are no matching option.

Example use case. Creating a service order for a customer account and having to select a contact under customer account.
In the service order form when creating, a customer account is selected, and then depending on the selected customer account, there is selection option to choose a contact under the account. Lets say there was no contact initially or the contact needs to be entered as a new record under selected customer account. In this case, if creating contact is allowed, a plus button next to the select input can prompt a modal showing the form to create the contact.

**Comment by danharrin** [comment]
#2057
